egs/thchs30/s5/local/dae/run_dae.sh

```diff
@@ -0,0 +1,154 @@
+   local/dae/make_fbank.sh --nj $nj --cmd "$train_cmd"  \
+   --addnoise-opt """$addnoise_opt"""  \
+   data/dae/train exp/dae/gendata fbank/dae/train || exit 1
+   steps/compute_cmvn_stats.sh data/dae/train exp/dae/cmvn \
+   fbank/dae/train/_cmvn || exit 1
+
+
+   # genreate dev data. Just the 0db condition is produced.  Multiple noise types mixed together.
+   sigma0=0 #no random in SNR
+   addnoise_opt="--noise-level $noise_level --sigma0 $sigma0 --seed $seed --verbose $verbose --noise-prior $noise_prior --noise-src $noise_scp --wavdir NULL"
+   rm -rf data/dae/dev/0db && mkdir -p data/dae/dev/0db && \
+   cp -L data/fbank/dev/{wav.scp,spk2utt,utt2spk,text} data/dae/dev/0db || exit 1
+   local/dae/make_fbank.sh --nj $nj --cmd "$train_cmd"  \
+   --addnoise-opt """$addnoise_opt"""  \
+   data/dae/dev/0db exp/dae/gendata fbank/dae/dev/0db || exit 1
+   steps/compute_cmvn_stats.sh data/dae/dev/0db exp/dae/cmvn \
+   fbank/dae/dev/0db/_cmvn || exit 1
+
+
+   # generate test data. Note that if you want to compare with the standard results, set stdtest=true
+   if [ $stdtest = true ]; then
+     #download noisy wav if use the standard test data
+     echo "using standard test data"
+     if [ $dwntest = true ];then
+       echo "downloading the noisy test data from Tsinghua Univ..."
+       echo "this may be slow for some connections, and is not very stable."
+       echo "you may want to try alternative mirror sites."
+       echo "check http://data.cslt.org/thchs30/challenges/asr.html"
+       (
+        mkdir -p wav && cd wav && \
+        wget http://data.cslt.org/thchs30-openslr/test.0db.tgz || exit 1
+        tar xvf test.0db.tgz || exit 1
+       )
+     fi
+     #generate fbank
+     for x in car white cafe; do
+       echo "producing fbanks for $x"
+       mkdir -p data/dae/test/0db/$x && \
+       cp -L data/fbank/test/{spk2utt,utt2spk,text} data/dae/test/0db/$x && \
+       awk '{print $1 " wav/test-noise/0db/'$x'/"$1".wav"}' data/fbank/test/wav.scp > data/dae/test/0db/$x/wav.scp || exit 1
+       steps/make_fbank.sh --nj $nj --cmd "$train_cmd"  \
+        data/dae/test/0db/$x exp/dae/gendata fbank/dae/test/0db/$x || exit 1
+     done
+
+   else
+     #generate test data randomly
+     echo "generating noisy test data randomly.."
+
+     for x in car white cafe; do
+       case $x in
+         car)
+            noise_prior="0.0,0.0,10.0,0.0" 
+            ;;
+         white)
+            noise_prior="0.0,10.0,0.0,0.0" 
+            ;;
+         cafe)
+            noise_prior="0.0,0.0,0.0,10.0" 
+            ;;
+       esac
+
+       addnoise_opt="--noise-level $noise_level --sigma0 $sigma0 --seed $seed --verbose $verbose --noise-prior $noise_prior --noise-src $noise_scp --wavdir NULL"
+       rm -rf data/dae/test/0db/$x && mkdir -p data/dae/test/0db/$x && \
+       cp -L data/fbank/test/{wav.scp,spk2utt,utt2spk,text} data/dae/test/0db/$x || exit 1
+       local/dae/make_fbank.sh --nj $nj --cmd "$train_cmd"  \
```

Comment:
I have a problem with the way you have integrated your addition of noise into the 'make_fbank.sh'.  It makes it impossible to do anything that needs non-fbank features.  And these recipes that you have used, nnet1 recipes based on filterbanks, are very old recipes; we have a lot of other things now that will probably work better.  Instead you should do this all at the wav.scp level, so you have a data directory that has commands inside the wav.scp to add the noise.  Or if that's not efficient, then dump noise-added data to disk and put their locations in the wav.scp.  Then it's a standard data directory and people can dump any kinds of features they want.